pandas/core/categorical.py

```diff
@@ -1851,7 +1851,7 @@ def unique(self):
         """
 
         # unlike np.unique, unique1d does not sort
-        unique_codes = unique1d(self.codes)
+        unique_codes = unique1d(self.codes).astype(self.codes.dtype)
```

Comment:
actually you don't need this change. This is just for taking (which needs to be int64 anyhow; it will upcast it later if its not).